dotcom-rendering/src/web/server/blocksToHtml.tsx

```diff
@@ -38,28 +41,37 @@ export const blocksToHtml = ({
 		adUnit,
 	});
 
+	const key = 'dcr';
+	const cache = createCache({ key });
+	const emotionServer = createEmotionServer(cache);
+
 	const html = renderToString(
-		<LiveBlogRenderer
-			blocks={blocks}
-			format={format}
-			adTargeting={adTargeting}
-			host={host}
-			pageId={pageId}
-			webTitle={webTitle}
-			ajaxUrl={ajaxUrl}
-			isSensitive={isSensitive}
-			isAdFreeUser={isAdFreeUser}
-			switches={switches}
-			isLiveUpdate={true}
-			section={section}
-			// The props below are never used because isLiveUpdate is true but, typescript...
-			shouldHideReaderRevenue={false}
-			tags={[]}
-			isPaidContent={false}
-			contributionsServiceUrl=""
-			keywordIds={keywordIds}
-		/>,
+		<CacheProvider value={cache}>
+			<LiveBlogRenderer
+				blocks={blocks}
+				format={format}
+				adTargeting={adTargeting}
+				host={host}
+				pageId={pageId}
+				webTitle={webTitle}
+				ajaxUrl={ajaxUrl}
+				isSensitive={isSensitive}
+				isAdFreeUser={isAdFreeUser}
+				switches={switches}
+				isLiveUpdate={true}
+				section={section}
+				// The props below are never used because isLiveUpdate is true but, typescript...
+				shouldHideReaderRevenue={false}
+				tags={[]}
+				isPaidContent={false}
+				contributionsServiceUrl=""
+				keywordIds={keywordIds}
+			/>
+		</CacheProvider>,
 	);
 
-	return html;
+	const chunks = emotionServer.extractCriticalToChunks(html);
+	const extractedCss = emotionServer.constructStyleTagsFromChunks(chunks);
```

Comment:
By adding a singleton for `emotionServer` could the cache creation logic be centralised and then used wherever it is required for the SSR?:

```suggestion
  const emotionServer = getEmotionServer();
	const chunks = emotionServer.extractCriticalToChunks(html);
	const extractedCss = emotionServer.constructStyleTagsFromChunks(chunks);
```